src/modules/Dashboard/features/Applet/Schedule/Schedule.hooks.ts

```diff
@@ -190,15 +193,31 @@ export const usePreparedEvents = (
   } = eventsData ?? {};
 
   useEffect(() => {
-    if (calendarEventsArr) {
-      dispatch(calendarEvents.actions.setCalendarEvents({ events: calendarEventsArr }));
+    if (
+      calendarEventsArr &&
+      !isEqual(calendarEventsArr, prevCalendarEventsArrRef.current) &&
+      (!processedEventStartYear || processedEventStartYear === currentYear)
+    ) {
+      dispatch(calendarEvents.actions.createCalendarEvents({ events: calendarEventsArr }));
+      prevCalendarEventsArrRef.current = calendarEventsArr;
     }
   }, [calendarEventsArr]);
 
   useEffect(() => {
-    if (eventsDataArr) {
-      dispatch(calendarEvents.actions.setCreateEventsData(eventsDataArr));
-    }
+    (async () => {
+      if (eventsDataArr && !isEqual(eventsDataArr, prevEventsDataArrRef.current)) {
```

Comment:
could yoiu add a variable for the conditions and use early return?

Reply:
done